main/static/js/eosidp.js

```diff
@@ -0,0 +1,20 @@
+// Facebook Login OAuth redirects the user with an ugly #_=_ URL
+// fragment. Strip that off.
+//
+// https://stackoverflow.com/a/18305085
+function stripFacebookOAuthFragment() {
+    if (window.location.hash === "#_=_") {
+        // If the browser supports history.replaceState, strip the whole
+        // fragment. Otherwise just the _=_.
+        if (history.replaceState) {
+            var noFragment = window.location.href.split("#")[0];
+            history.replaceState(null, null, noFragment);
+        } else {
+            window.location.hash = "";
+        }
+    }
+}
+
+// Strip the fragment immediately and also whenever it changes.
```

Comment:
I would use `window.onload` in this case. Is there a point in repeatedly removing the hash afterwards?

Reply:
I did have that but thought just stripping the fragment didn't need to wait until the entire page loaded, though. I think you're probably right, though.

As for when the hash changes... I can't really think of a good reason. It was helpful when testing it out, but it probably adds overhead when following anchors. I think I'll remove that.